syncer/syncer.go

```diff
@@ -1837,6 +1837,10 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext, o
 			case *ast.TruncateTableStmt:
 				ec.tctx.L().Info("ignore truncate table statement in shard group", zap.String("event", "query"), zap.String("statement", sqlDDL))
 				continue
+			// TODO: Support rename a table to the same shard group
+			// e.g. upstream: tb* -> downstream: tb, rename tb1 to tb2
+			case *ast.RenameTableStmt:
```

Comment:
will it error for the non-sharding tables?

Reply:
Yes, it does happen with the above implementation. I'm working on “rename to the same group" and will try to fix it.

Reply:
Actually, if mode is pessimistic, there isn't non-sharding tables.

Reply:
OK, that's related to https://github.com/pingcap/dm/issues/1517. In future we should support specify part of sources to form a sharding group.